charms/kserve-controller/tests/integration/test_charm.py

```diff
@@ -51,45 +51,24 @@
 }
 
 
-@tenacity.retry(
-    wait=tenacity.wait_exponential(multiplier=2, min=1, max=10),
-    stop=tenacity.stop_after_attempt(80),
-    reraise=True,
-)
-def assert_deleted(logger, client, resource_class, resource_name, namespace):
-    """Test for deleted resource. Retries multiple times to allow deployment to be deleted."""
-    logger.info(f"Waiting for {resource_class}/{resource_name} to be deleted.")
-    deleted = False
-    try:
-        dep = client.get(resource_class, resource_name, namespace=namespace)
-        state = dep.get("status", {}).get("state")
-    except ApiError as error:
-        logger.info(f"Not found {resource_class}/{resource_name}. Status {error.status.code} ")
-        if error.status.code == 404:
-            deleted = True
-
-    assert deleted, f"Waited too long for {resource_class}/{resource_name}:{state} to be deleted!"
-
-
```

Comment:
We actually want to keep this helper method and call it in the `cleanup_namespaces_after_execution` fixture to avoid issues with resources.